### PR TITLE
add linting for dashboard visualization code

### PR DESCRIPTION
**Pull request by imatiach-msft**
Adding linting to the visualization dashboard - based on this guide:
https://www.robertcooper.me/using-eslint-and-prettier-in-a-typescript-project

### CODE REVIEW

**Comment by imatiach-msft** [comment]
any idea why build-webpage is failing?  I can't seem to access the error.

**Comment by kevinrobinson** [comment]
Would it be useful to coordinate with
https://github.com/fairlearn/fairlearn/pull/283 before merging this?

**Comment by kevinrobinson** [comment]
Info on the build error is in
https://github.com/fairlearn/fairlearn/pull/490#issuecomment-646283360 with
a link to the fix from @riedgar-ms too.

**Comment by riedgar-ms** [comment]
My change is in, but for some reason the docbuild keeps failing. Can you try merging in the changes from master, and see if that fixes it?

**Comment by riedgar-ms** [comment]
Thank you for adding the linter. I'll defer to @rihorn2  for signoff, though.

**Comment by imatiach-msft** [comment]
@kevinrobinson good point, if this needs to be coordinated I can wait.  The linting changes are mostly automatic and can be re-run on the new PR.

**Comment by riedgar-ms** [comment]
@rihorn2 , can you comment wrt the linked #283 ?

File: visualization/FairnessDashboard/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    rules: {
+        '@typescript-eslint/no-this-alias': 'off',
+        'react/display-name': 'off',
+        '@typescript-eslint/no-empty-function': 'off',
+        '@typescript-eslint/no-empty-interface': 'off',
+        '@typescript-eslint/no-var-requires': 'off',
+        'react/prop-types': 'off',
+        'react/jsx-key': 'off',
+        'react/no-unescaped-entities': 'off',
+        'react/no-direct-mutation-state': 'off',
```
Comment (rihorn2) [inline visualization/FairnessDashboard/.eslintrc.js:30]: I'm guessing these are all teh places that raised errors when run, and so for now are turned off? most seem fine, we can try to enable some in a future pr

Comment (imatiach-msft) [reply visualization/FairnessDashboard/.eslintrc.js:30]: yes, exactly

**Comment by imatiach-msft** [comment]
FYI: Coordinated with Chinmay and he said it is ok to merge this PR.
@romanlutz can this PR be merged (I don't have the permissions)?

**Pull request merged**